migrations/1618843631500-users.js

```diff
@@ -0,0 +1,90 @@
+var db = require('./db')
+
+exports.up = function (next) {
+  var sql = [
+    `CREATE TYPE role AS ENUM('user', 'superuser')`,
+    `CREATE TABLE users (
+      id UUID PRIMARY KEY,
+      username VARCHAR(50) NOT NULL UNIQUE,
+      password VARCHAR(100),
+      role role NOT NULL DEFAULT 'user',
+      enabled BOOLEAN DEFAULT true,
+      twofa_code VARCHAR(100),
+      temp_twofa_code VARCHAR(100),
+      created TIMESTAMPTZ NOT NULL DEFAULT now(),
+      last_accessed TIMESTAMPTZ NOT NULL DEFAULT now(),
+      last_accessed_from TEXT,
+      last_accessed_address INET )`,
+    `CREATE TABLE "user_sessions" (
+      "sid" VARCHAR NOT NULL COLLATE "default",
+      "sess" JSON NOT NULL,
+      "expire" TIMESTAMP(6) NOT NULL )
+      WITH (OIDS=FALSE)`,
+    `ALTER TABLE "user_sessions" ADD CONSTRAINT "session_pkey" PRIMARY KEY ("sid") NOT DEFERRABLE INITIALLY IMMEDIATE`,
+    `CREATE INDEX "IDX_session_expire" ON "user_sessions" ("expire")`,
+    `CREATE TYPE auth_token_type AS ENUM('reset_password', 'reset_twofa')`,
+    `CREATE TABLE auth_tokens (
+      token TEXT NOT NULL PRIMARY KEY,
+      type auth_token_type NOT NULL,
+      user_id UUID REFERENCES users(id) ON DELETE CASCADE,
+      expire TIMESTAMPTZ NOT NULL DEFAULT now() + interval '30 minutes',
+      CONSTRAINT unique_userid_type UNIQUE (user_id, type)
+    )`,
+    `CREATE TABLE user_register_tokens (
+      token TEXT NOT NULL PRIMARY KEY,
+      username TEXT NOT NULL UNIQUE,
+      role role DEFAULT 'user',
+      expire TIMESTAMPTZ NOT NULL DEFAULT now() + interval '30 minutes'
+    )`,
+    // migrate values from customers which reference user_tokens for data persistence
+    `ALTER TABLE customers ADD COLUMN sms_override_by_old TEXT`,
```

Comment:
This kind of pollutes the table. Is there a way to keep this information in the new schema? If not maybe it should be in a separate table.

Reply:
To solve this, I created two temporary tables whose purpose is to store this old data while the gradual transition to authenticated users occurs